sno/gpkg_adapter.py

```diff
@@ -0,0 +1,246 @@
+        "definition": "",
+        "organization": "EPSG",
+        "srs_id": 0,
+        "organization_coordsys_id": 0,
+        "description": None,
+    }
+]
+
+
+def wkt_to_gpkg_spatial_ref_sys(wkt):
+    """Given a WKT srs definition, generate a gpkg_spatial_ref_sys meta item."""
+    return _gpkg_spatial_ref_sys(SpatialReference(wkt), wkt)
+
+
+def osgeo_to_gpkg_spatial_ref_sys(spatial_ref):
+    """Given an osgeo SpatialReference, generate a gpkg_spatial_ref_sys meta item."""
+
+    return _gpkg_spatial_ref_sys(spatial_ref, spatial_ref.ExportToWkt())
+
+
+def _gpkg_spatial_ref_sys(spatial_ref, wkt):
+    spatial_ref.AutoIdentifyEPSG()
+    return [
+        {
+            "srs_name": spatial_ref.GetName(),
+            "definition": wkt,
+            "organization": spatial_ref.GetAuthorityName(None),
+            "srs_id": spatial_ref.GetAuthorityCode(None),
+            "organization_coordsys_id": spatial_ref.GetAuthorityCode(None),
+            "description": None,
+        }
+    ]
+
+
+DEFAULT_SRS_STR = "EPSG:0"
+
+
+def wkt_to_srs_str(wkt):
+    """Given a WKT srs definition, generate a sensible name for it."""
+    return osgeo_to_srs_str(SpatialReference(wkt))
+
+
+def osgeo_to_srs_str(spatial_ref):
+    """Given a osgeo SpatialReference, generate a sensible name for it."""
+    auth_name = spatial_ref.GetAuthorityName(None)
+    auth_code = spatial_ref.GetAuthorityCode(None)
+    return f"{auth_name}:{auth_code}"
+
+
+def generate_sqlite_table_info(dataset):
+    """Generate a sqlite_table_info meta item from a dataset."""
+    return [_columnschema_to_gpkg(i, col) for i, col in enumerate(dataset.schema)]
+
+
+def _get_geometry_columns(schema):
+    return [c for c in schema.columns if c.data_type == "geometry"]
+
+
+def gpkg_to_v2_schema(
+    sqlite_table_info, gpkg_geometry_columns, gpkg_spatial_ref_sys, id_salt
+):
+    """Generate a v2 Schema from the given gpkg meta items."""
+    return Schema(
+        [
+            _gpkg_to_column_schema(
+                col, gpkg_geometry_columns, gpkg_spatial_ref_sys, id_salt
+            )
+            for col in sorted(sqlite_table_info, key=_sort_by_cid)
+        ]
+    )
+
+
+def _sort_by_cid(sqlite_col_info):
+    return sqlite_col_info["cid"]
+
+
+def _gpkg_to_column_schema(
+    sqlite_col_info, gpkg_geometry_columns, gpkg_spatial_ref_sys, id_salt
+):
+    name = sqlite_col_info["name"]
+    pk_index = 0 if sqlite_col_info["pk"] == 1 else None
+    if gpkg_geometry_columns and name == gpkg_geometry_columns["column_name"]:
+        data_type, extra_type_info = _gkpg_geometry_columns_to_v2_type(
+            gpkg_geometry_columns, gpkg_spatial_ref_sys,
+        )
+    else:
+        data_type, extra_type_info = gpkg_type_to_v2_type(sqlite_col_info["type"])
+
+    col_id = ColumnSchema.deterministic_id(name, data_type, id_salt)
+    return ColumnSchema(col_id, name, data_type, pk_index, **extra_type_info)
+
+
+def _columnschema_to_gpkg(cid, column_schema):
```

Comment:
should use `columnschema` or `column_schema` consistently (also see _gpkg_to_column_schema )

Reply:
Done